src/include/storage/irc_authorization.hpp

```diff
@@ -11,6 +11,8 @@ enum class IcebergEndpointType : uint8_t { AWS_S3TABLES, AWS_GLUE, INVALID };
 
 enum class IRCAuthorizationType : uint8_t { OAUTH2, SIGV4, NONE, INVALID };
 
+enum class IRCAccessDelegationMode : uint8_t { NONE, VENDED_CREDENTIALS };
```

Comment:
I think we want to introduce a `IRCAccessDelegationMode::DEFAULT` setting, and solidify this once we have the response from `v1/config`, which I believe tells us whether the catalog supports vended credentials.

When they do, `DEFAULT` becomes `VENDED_CREDENTIALS`, when they don't, it becomes `NONE`
This way we reduce the need to reach for ATTACH options, only when you're going against the grain of the Catalog (setting to NONE when the catalog *does* accept vended credentials)

We also want to verify whether the preference can be honored after we have the `v1/config` response

Reply:
> data-access:
      name: X-Iceberg-Access-Delegation
      in: header
      description: >
        Optional signal to the server that the client supports delegated access
        via a comma-separated list of access mechanisms.  The server may choose
        to supply access via any or none of the requested mechanisms.

When I read this, there shouldn't even be a need to omit this, the catalog is free to not honor the request..
It's merely saying that the client (duckdb-iceberg) supports it..